src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -1168,5 +1171,52 @@ public static IGraphQLBuilder AddExecutionStrategy<TExecutionStrategy>(this IGra
             return builder;
         }
         #endregion
+
+        #region - UseTelemetry -
+#if NET5_0_OR_GREATER
+        /// <summary>
+        /// Configures the GraphQL server to collect traces via the <see cref="System.Diagnostics.Activity">System.Diagnostics.Activity API</see> and records events that match the
+        /// <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/graphql.md">OpenTelemetry recommendations</see>.
+        /// Trace data contains the GraphQL operation name, the operation type, and the optionally the document.
+        /// </summary>
+        /// <remarks>
+        /// When applicable, place after calls to UseAutomaticPersistedQueries to ensure that the query document is recorded properly.
+        /// </remarks>
+        public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, ActivitySource activitySource, Action<TelemetryOptions>? configure = null)
+            => UseTelemetry(builder, _ => activitySource, configure != null ? (opts, _) => configure(opts) : null);
+
+        /// <inheritdoc cref="UseTelemetry(IGraphQLBuilder, ActivitySource, Action{TelemetryOptions}?)"/>
+        public static IGraphQLBuilder UseTelemetry(this IGraphQLBuilder builder, Func<IServiceProvider, ActivitySource> activitySourceFactory, Action<TelemetryOptions, IServiceProvider>? configure = null)
+        {
+            if (activitySourceFactory == null)
+                throw new ArgumentNullException(nameof(activitySourceFactory));
+
+            builder.Services.Configure(configure);
+            builder.ConfigureExecution(async (options, next) =>
+            {
+                var activitySource = activitySourceFactory(options.RequestServices ?? throw new MissingRequestServicesException());
```

Comment:
In general, `ActivitySource` should be created once, stored in a static variable, and reused by the library ([source](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Api/README.md#basic-usage)). The library authors should decide on the activity source name, not the end users. It's also recommended to provide the version to the `ActivitySource` constructor.

Reply:
Sounds good; I'll make the changes.